java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java

```diff
@@ -117,6 +117,10 @@ public boolean isEmpty() {
   }
 
   public static AnalyzerMessage.TextSpan textSpanFor(Tree syntaxNode) {
+    if (syntaxNode.is(Tree.Kind.TEXT_BLOCK)) {
+      SyntaxToken syntaxToken = getNonEmptyTree(syntaxNode).firstToken();
+      return textSpanForTextBlock(syntaxToken);
```

Comment:
if we already know it's a `TEXT_BLOCK`, I feel we could bypass the `getNonEmptyTree()` and directly call:
```java
return textSpanForTextBlock(((LiteralTree) syntaxNode).token());
```